### DISCUSSION TITLE
Things I noticed after using AdminJS for a while

### DISCUSSION THREAD

**Discussion by DiogoAbu** [SoftwareBrothers/admin-bro#1439] [Ideas]
- [ ] The `phone` property type does not format when displaying, only when editing with `PhoneInput`. I used [google-libphonenumber](https://github.com/ruimarinho/google-libphonenumber) for this.
- [ ] The `phone` property type does not have translation for `search` and `searchNotFound` on `PhoneInput`.
- [ ] The `datetime` property type is also missing translations, on edit, and formatting according to locale, on edit and display.
- [ ] The `CurrentAdmin` typing should be changed from `type` to `interface` so it can be overridden.
- [ ] Some property types does not respect the `isRequired` option: currency, datetime, default-type (select), password, text-area.
- [ ] When a form fails the validation and the errors appears, should we clear the errors after the user changes the input? I set the error to a state and then clear it on change.
- [ ] The labels on `useNavigationResources`, which are the resources names, should be translated.
- [ ] The `availableValues` labels are translated on the backend which does not change locale when changing on the frontend, currently i'm getting the locale on the frontend using `navigator.language` and using a express middleware to change the language based on the `accept-language` header.

I have this on my fork (among some custom stuff) and would like to know if it is already being done for v7 or should I pr this things?

**Comment by dziraf** [comment]
> The phone property type does not format when displaying, only when editing with PhoneInput. I used [google-libphonenumber](https://github.com/ruimarinho/google-libphonenumber) for this.

I guess currently it's just a simple text component, feel free to create a PR for this since it's not a breaking change.

> The phone property type does not have translation for search and searchNotFound on PhoneInput.

Responded to your PR.

> The datetime property type is also missing translations, on edit, and formatting according to locale, on edit and display.

This might have to be translated in the design system which is currently not possible but possible in v7. We'll see, it should be fixed in either 7.0.0 or some patch version later.

>  The CurrentAdmin typing should be changed from type to interface so it can be overridden.

Feel free to open a PR, we can also do it alongside other changes in v7.

>  Some property types does not respect the isRequired option: currency, datetime, default-type (select), password, text-area.

Select should be fixed in v7 core/v4 design system, we'll take a look at other ones.

> When a form fails the validation and the errors appears, should we clear the errors after the user changes the input? I set the error to a state and then clear it on change.

Not sure what's the best approach here; if you have a lot of errors you might want them to stay until you submit I guess?

> The labels on useNavigationResources, which are the resources names, should be translated.

We moved all translations to the frontend in v7, will check if it's fixed there.

> The availableValues labels are translated on the backend which does not change locale when changing on the frontend, currently i'm getting the locale on the frontend using navigator.language and using a express middleware to change the language based on the accept-language header.

As above, we no longer translate anything in the backend (it only sends translation codes where needed), will see if it's fixed.

**Reply by DiogoAbu** [reply]
Not translating on the backend was the right move, I was gonna suggest it... 

About clearing the errors, it would clear one at a time, just the ones for the specific input the users changes... but clearing all on submit would work too.

I'll take a look at opening PRs for the other things.
